include/circt/Dialect/Calyx/CalyxInterfaces.td

```diff
@@ -91,9 +91,15 @@ def CellOpInterface : OpInterface<"CellInterface"> {
       /*methodBody=*/"",
       /*defaultImplementation=*/[{
         SmallVector<circt::calyx::PortInfo> info;
-        for (auto it : llvm::zip($_op->getResults(), $_op.portDirections(),$_op.portNames()))
-          info.push_back(PortInfo{StringAttr::get($_op->getContext(),
-            std::get<2>(it)), std::get<0>(it).getType(), std::get<1>(it)});
+
+        auto zip = llvm::zip($_op->getResults(), $_op.portDirections(), $_op.portNames());
+        for (auto&& [result, direction, name] : zip)
+          info.push_back(PortInfo{
+              .name=StringAttr::get($_op->getContext(), name),
```

Comment:
I think naming the struct members in the initializer like this is a C++20 feature (and we're on 17 for now), do you get a warning about that?

Reply:
No warnings AFAICT. I've been doing this for a at least a year or two now. Here's the [cppreference](https://en.cppreference.com/w/c/language/struct_initialization) on struct initialization, if I'm understanding correctly what you're referencing:

Reply:
Ok great. I was thinking of "designated initializers" (syntax (3) and (4) here: https://en.cppreference.com/w/cpp/language/aggregate_initialization).